src/fondant/components/retrieve_from_faiss_by_prompt/src/main.py

```diff
@@ -47,11 +48,27 @@ def __init__(  # PLR0913
 
     def setup(self) -> Client:
         """Setup LocalCudaCluster if gpu is available."""
+        dask.config.set({"dataframe.convert-string": False})
+        dask.config.set({"distributed.worker.daemon": False})
+
         if self.device == "cuda":
             cluster = LocalCUDACluster()
             return Client(cluster)
 
-        return super().setup()
+        total_memory = (os.sysconf("SC_PAGE_SIZE") * os.sysconf("SC_PHYS_PAGES")) / (
+            1024**3
+        )
+        # We need at least 8Gb RAM for the datacomp small faiss index
+        # We should consider calculating the memory required for the index based on the faiss
+        # index size
+        cores_to_utilise = int(total_memory / 8)
```

Comment:
```suggestion
        cores_to_utilise = total_memory // 8
```

You probably want to floor it to have at least 8GB per worker.